### DISCUSSION TITLE
How to import remark plugins (remark-gfm) in next.config.js?

### DISCUSSION THREAD

**Discussion by linkai101** [vercel/next.js#36310] [Help] [answered]
I basically have the same question as this [stack overflow question](https://stackoverflow.com/questions/71576438/how-to-import-remark-gfm-in-next-config-js) and I haven't been able to find any answers on any forum or docs.

I'm working with @next/mdx and I want to use the remark-gfm plugin, but I'm not sure how to import it into next.config.js as it suggests in the [Next.js Docs on MDX](https://nextjs.org/docs/advanced-features/using-mdx#setup-nextmdx-in-nextjs).

```js
// next.config.js
const withMDX = require('@next/mdx')({
  extension: /\.mdx?$/,
  options: {
    remarkPlugins: [], // how to import remark-gfm?
    rehypePlugins: [],
    providerImportSource: "@mdx-js/react",
  },
});

module.exports = withMDX({
  reactStrictMode: true,
  pageExtensions: ['ts', 'tsx', 'js', 'jsx', 'md', 'mdx'],
});

```

I've tried `import`, `require`, and plain text in the remarkPlugins array in next.config.js to import the npm package but they've all produced errors. Anyone know how to do it?

**Comment by icyJoseph** [comment, marked as answer]
Rename your `next.config.js` to `next.config.mjs` and do like this:

```js
// next.config.mjs
import remarkGfm from "remark-gfm";
import createMDX from "@next/mdx";

/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  pageExtensions: ["js", "jsx", "ts", "tsx", "md", "mdx"],
};

const withMDX = createMDX({
  extension: /\.mdx?$/,
  options: {
    remarkPlugins: [remarkGfm],
    rehypePlugins: [],
  },
});

export default withMDX(nextConfig);
```

**Reply by linkai101** [reply]
That worked, thanks!

**Reply by max8hine** [reply]
Quoted from https://nextjs.org/docs/advanced-features/using-mdx

>  If you use remark-gfm, you'll need to use next.config.mjs

**Reply by nenadV91** [reply]
Thanks so much for this 🙏

**Reply by jonxuxu** [reply]
Thanks, that got gfm working for me too! But this caused another issue, only on mdx pages that use gfm components. Does the package create html components in a way that breaks server side rendering?
```
Unhandled Runtime Error
Error: Hydration failed because the initial UI does not match what was rendered on the server.

Warning: Expected server HTML to contain a matching <center> in <p>.

See more info here: https://nextjs.org/docs/messages/react-hydration-error
Component Stack
Call Stack
performConcurrentWorkOnRoot
```

Has anyone else had this issue?
